pkg/controlplane/instance.go

```diff
@@ -144,7 +145,7 @@ func (cp *Instance) DeletePeer(name string) (*cpstore.Peer, error) {
 		return nil, err
 	}
 	if peer == nil {
-		return nil, nil
+		return nil, errors.New("peer not found")
```

Comment:
the current behavior for deleting an object was to return the object (first return) and an error (second return).
When trying to remove an object that does not exist, the code returned `nil` and no error (since the user's action was satisfied - the object is not present).

I wonder if the callers are relying on this behavior (nil, nil) and would choke on the non-nil error return.
Have you confirmed the call site's behavior with a non-nil error?

Reply:
@orozery - It looks likt it is not breaking anything - can you please confirm?

Reply:
@elevran is right.
The caller in this case is:
```
result, err := spec.Handler.Delete(object)
	if err != nil {
		requestLogger.Errorf("Cannot delete object: %v.", err)
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	if reflect.ValueOf(result).IsNil() {
		w.WriteHeader(http.StatusNotFound)
		return
	}
```

The proposed change will cause an HTTP500 (internal server error) instead of HTTP404 (not found) for a non-existing peer.

Reply:
Perhaps we should explicitly define a "not found" error and test for it in caller code to return 404?
```go
// in pkg/controlplane/store
var ErrNotFound = errors.New("Resource was not found")
...

// in all `return nil, nil` locations:
... 
  return nil, ErrNotFound
...

// in caller sites:
        ...
        obj, err := spec.Handler.Delete(object)
   
       if err != nil && !errors.Is(err, store.ErrNotFound) {
		requestLogger.Errorf("Cannot delete object: %v.", err)
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	if reflect.ValueOf(result).IsNil() {
		w.WriteHeader(http.StatusNotFound)
		return
	}
}
```
See https://pkg.go.dev/errors for discussion of error.Is.